mne_bids/mne_bids.py

```diff
@@ -619,6 +644,33 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
                                      output_path=output_path, make_dir=False,
                                      overwrite=False, verbose=verbose)
 
+    processing = None
+    proc_history = raw.info['proc_history']
+    if proc_history != []:
+        sss = False
+        tsss = False
+        for ph in proc_history:
+            max_info = ph.get('max_info')
+            if max_info:
+                if (max_info['sss_info'] != dict() or
+                        max_info['sss_ctc'] != dict() or
+                        max_info['sss_cal'] != dict()):
+                    sss = True
+                if max_info['max_st'] != dict():
+                    tsss = True
+        if tsss:
+            processing = 'tsss'
+        elif sss:
+            processing = 'sss'
+
+    # regenerate the bids_basename with the processing info if it is required
+    if processing is not None:
+        bids_basename = make_bids_basename(
+            subject=subject_id, session=session_id, task=task,
+            acquisition=acquisition, run=run, processing=processing,
+            space=params['space'], recording=params['recording'])
```

Comment:
I would remove this. Because files with this naming should be placed in the "derivatives" folder according to the BIDS specification. That is not what we support currently. Also, it's surprising behavior if you change the filename that users provide ...

Reply:
Has it actually been decided that sss/tsss processed data is to go in the derivatives folder? (I thought there was still ambiguity in the specification...)
After a discussion I had with some others who use Elekta data they agreed that the sss/tsss processed data is pretty much the "rawest" usable data possible.

Reply:
I think currently that is what the specification says and what the BIDS-example say. See here: https://github.com/OpenNeuroDatasets/ds000117/tree/master/derivatives/meg_derivatives/sub-01/ses-meg ...

Reply:
ok, fair enough. I will put it in the derivatives folder.
However my point still stands that we need to add the `proc-sss` or proc-tsss` to the file name. Unless you think it actually better to just expect the user to provide it in the path name?

Reply:
Let's give this some more thought before rushing to a decision.

I think it's not so unreasonable to expect users to provide the filename. Because, by default MaxFilter appends `_sss.fif` or `_tsss.fif` to the filename, they will know even without inspecting the raw object that the filename should have `proc-sss`. We could throw an error if it's processed by SSS but the filename does not contain this segment.

I'm not exactly sure about the "derivatives" thing. Whatever we do is likely to change later as the derivatives spec is not ready. We are waiting for EEG and iEEG specs to be merged first. But this is probably at least a few months away from being formalized.